src/components/App.tsx

```diff
@@ -344,30 +344,37 @@ class App extends React.Component<any, AppState> {
       /^#json=([0-9]+),([a-zA-Z0-9_-]+)$/,
     );
 
-    const isCollaborationScene = getCollaborationLinkData(window.location.href);
-
-    if (!isCollaborationScene) {
-      let scene: ResolutionType<typeof loadScene> | undefined;
-      // Backwards compatibility with legacy url format
-      if (id) {
-        scene = await loadScene(id);
-      } else if (jsonMatch) {
-        scene = await loadScene(jsonMatch[1], jsonMatch[2]);
+    let isCollaborationScene = getCollaborationLinkData(window.location.href);
+    let scene = await loadScene(null);
+
+    if (id || jsonMatch || isCollaborationScene) {
```

Comment:
can you create a variable `loadingExternalScene` and assign `!!(id || jsonMatch || isCollaborationScene)` ? so its clear that this condition is to check if we are loading external scene